solo-chains/runtime/starlight/src/lib.rs

```diff
@@ -1606,6 +1611,76 @@ parameter_types! {
     pub const MaxLengthParaIds: u32 = 100u32;
     pub const MaxEncodedGenesisDataSize: u32 = 5_000_000u32; // 5MB
 }
+
+pub struct InnerStarlightRegistrar<Runtime, AccountId, RegistrarManager, RegistrarWeightInfo>(
+    PhantomData<(Runtime, AccountId, RegistrarManager, RegistrarWeightInfo)>,
+);
+impl<Runtime, AccountId, RegistrarManager, RegistrarWeightInfo> RegistrarHandler<AccountId>
+    for InnerStarlightRegistrar<Runtime, AccountId, RegistrarManager, RegistrarWeightInfo>
+where
+    RegistrarManager: RegistrarInterface<AccountId = AccountId>,
+    RegistrarWeightInfo: paras_registrar::WeightInfo,
+    Runtime: pallet_registrar::Config,
+{
+    fn register(
+        who: AccountId,
+        id: ParaId,
+        genesis_storage: Vec<ContainerChainGenesisDataItem>,
+        head_data: Option<HeadData>,
+    ) -> DispatchResult {
+        // Return early if head_data is not specified
+        let genesis_head = match head_data {
+            Some(data) => data,
+            None => return Err(ContainerRegistrarError::<Runtime>::HeadDataNecessary.into()),
+        };
+
+        let key_values: Vec<(Vec<u8>, Vec<u8>)> =
+            genesis_storage.into_iter().map(Into::into).collect();
```

Comment:
You don't need to collect here if you just need to find the code, try something like this:

```rust
// Check if the wasm code is present in storage
let validation_code = match genesis_storage
    .into_iter()
    .map(Into::into)
    .find(|(key, _): &(Vec<u8>, Vec<u8>)| key == &StorageWellKnownKeys::CODE)
{
    Some((_, code)) => ValidationCode(code),
    None => return Err(ContainerRegistrarError::<Runtime>::WasmCodeNecessary.into()),
};
```

Also don't call `CODE.to_vec()` on every iteration, you should be able to compare references

Reply:
Thanks for the suggestion! 

Fixed in 73daab1933b5cfc51de53b7c7e8512420e7907c0.